daprdocs/content/en/go-sdk-docs/go-advanced/_index.md

```diff
@@ -0,0 +1,74 @@
+---
+type: docs
+title: "Advanced uses of the Dapr pluggable components .Go SDK"
+linkTitle: "Advanced"
+weight: 2000
+description: How to use advanced techniques with the Dapr pluggable components Go SDK
+is_preview: true
+---
+
+While not typically needed by most, these guides show advanced ways you can configure your Go pluggable components.
+
+## Component lifetime
+
+Components are registered by passing a "factory method" that is called for each configured component of that type associated with that socket. The method returns the instance associated with that component (whether shared or not). This allows multiple components of the same type to be configured with different sets of metadata, when component operations need to be isolated from one another, etc.
+
+## Registering multiple services
+
+Each call to `Register()` binds a socket to a registered component. One of each component type (input/output binding, pub/sub, and state store) can be registered per socket.
+
+```go
+func main() {
+	dapr.Register("service-a", dapr.WithStateStore(func() state.Store {
+		return &components.MyDatabaseStoreComponent{}
+	}))
+
+	dapr.Register("service-a", dapr.WithOutputBinding(func() bindings.OutputBinding {
+		return &components.MyDatabaseOutputBindingComponent{}
+	}))
+
+    dapr.Register("service-b", dapr.WithStateStore(func() state.Store {
+		return &components.MyDatabaseStoreComponent{}
+	}))
+
+	dapr.MustRun()
+}
+```
+
```

Comment:
Should we make a call out that the above code defines 2 components `service-a`  and `service-b`, the first hosting 2 components while the latter hosts only 1? This would make it simpler for the reader to tie how this is linked to the 2 configurations in the section bellow and with the 2 UDS files that are created.

Reply:
I added some verbiage to (hopefully) help tie the two sections together.